src/app/components/sandbox-controller/sandbox-controller.component.spec.ts

```diff
@@ -1,29 +1,42 @@
+import { cold } from 'jasmine-marbles';
 import { MockComponent, MockService } from 'ng-mocks';
+import { of } from 'rxjs';
 import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
-import { ReactiveFormsModule } from '@angular/forms';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
 import { MatCardModule } from '@angular/material/card';
 import { MatIconModule } from '@angular/material/icon';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { RouterTestingModule } from '@angular/router/testing';
 import { CodemirrorComponent } from '@ctrl/ngx-codemirror';
-import { Example, EXAMPLE, ExampleSection, START_EXAMPLE } from '../../core';
-import { ExecutorService } from '../../services';
+import { Example, EXAMPLE, InputStream, InputStreamLike, OutputStream, START_EXAMPLE } from '../../core';
+import { ExecutorService, RuntimeService, StreamBuilderService } from '../../services';
 import { StreamControllerComponent } from '../stream-controller/stream-controller.component';
 import { SandboxControllerComponent } from './sandbox-controller.component';
 
-class MockExample implements Example {
-  name = 'MockExample';
-  section: ExampleSection = 'combination';
-
-  getInputStreams = () => ({ large: [], small: [] });
-  getCode = () => '';
-}
-
 describe('SandboxControllerComponent', () => {
   let component: SandboxControllerComponent;
   let fixture: ComponentFixture<SandboxControllerComponent>;
 
+  let executorServiceSpy: jasmine.SpyObj<ExecutorService>;
+  let streamBuilderServiceSpy: jasmine.SpyObj<StreamBuilderService>;
+  let inputStreamSpy: jasmine.SpyObj<InputStreamLike>;
+  let exampleSpy: jasmine.SpyObj<Example>;
+
   beforeEach(waitForAsync(() => {
+    inputStreamSpy = jasmine.createSpyObj<InputStreamLike>('InputStreamLike', ['updateNode']);
```

Comment:
Name argument not really needed,

```suggestion
    inputStreamSpy = jasmine.createSpyObj<InputStreamLike>(['updateNode']);
```

Reply:
Leaving as is. I have done this in too many places and changing it doesn't make any real difference.
Noted for future PRs :)